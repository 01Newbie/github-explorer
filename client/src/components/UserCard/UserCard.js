import React from "react";

const UserCard = ({ user }) => {
  return (
    <div>
      {/* Render the user's name */}
      <h2>{user.name}</h2>
      {/* Display the user's email */}
      <p>{user.email}</p>

      {/* Check if the user has any repositories */}
      {user.repos && user.repos.length > 0 ? (
        <div>
          {/* Render the section header for repositories */}
          <h3>Repositories:</h3>
          {/* Iterate through each repository and display its details */}
          {user.repos.map((repo, index) => (
            <div key={index}>
              {/* Render the repository name */}
              <h4>{repo.name}</h4>
              {/* Display the repository description */}
              <p>{repo.description}</p>
              {/* Show the creation date of the repository */}
              <p>Created on: {repo.creationDate}</p>
              {/* Show the date of the last commit */}
              <p>Last commit: {repo.lastCommitDate}</p>
              {/* Display last commits if they exist */}
              {repo.lastCommits && repo.lastCommits.length > 0 && (
                <div>
                  {/* Render the section header for last commits */}
                  <h5>Last Commits:</h5>
                  <ul>
                    {/* Iterate through the last commits and display each one */}
                    {repo.lastCommits.map((commit, commitIndex) => (
                      <li key={commitIndex}>{commit}</li>
                    ))}
                  </ul>
                </div>
              )}
            </div>
          ))}
        </div>
      ) : (
        // Optional message for no repositories available
        <p>No repositories available.</p>
      )}
    </div>
  );
};

export default UserCard;
