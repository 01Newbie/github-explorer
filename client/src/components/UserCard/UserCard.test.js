import React from "react";
import { render, screen } from "@testing-library/react";
import UserCard from "./UserCard";

describe("UserCard", () => {
  // Test for rendering user name and email
  test("renders user name and email", () => {
    // Create a mock user object with name and email
    const user = { name: "John Doe", email: "john.doe@example.com" };

    // Render the UserCard component with the mock user
    render(<UserCard user={user} />);

    // Assert that the user name and email are present in the document
    expect(screen.getByText(/john doe/i)).toBeInTheDocument();
    expect(screen.getByText(/john.doe@example.com/i)).toBeInTheDocument();
  });

  // Test for rendering repository details
  test("renders repository details", () => {
    // Create a mock user object with repository details
    const user = {
      name: "John Doe",
      email: "john.doe@example.com",
      repos: [
        {
          name: "Repo1",
          description: "First repo",
          creationDate: "2024-01-01",
          lastCommitDate: "2024-01-10",
          lastCommits: ["Initial commit", "Updated README", "Fixed bugs"],
        },
      ],
    };

    // Render the UserCard component with the mock user
    render(<UserCard user={user} />);

    // Assert that the repository details are present in the document
    expect(screen.getByText(/repo1/i)).toBeInTheDocument();
    expect(screen.getByText(/first repo/i)).toBeInTheDocument();
    expect(screen.getByText(/2024-01-01/i)).toBeInTheDocument();
    expect(screen.getByText(/2024-01-10/i)).toBeInTheDocument();
  });

  // Snapshot test for UserCard
  test("UserCard snapshot", () => {
    // Create a mock user object with repository details for the snapshot
    const user = {
      name: "John Doe",
      email: "john.doe@example.com",
      repos: [
        {
          name: "Repo1",
          description: "First repo",
          creationDate: "2024-01-01",
          lastCommitDate: "2024-01-10",
          lastCommits: ["Initial commit", "Updated README", "Fixed bugs"],
        },
      ],
    };

    // Render the UserCard component and capture the rendered output
    const { asFragment } = render(<UserCard user={user} />);

    // Assert that the rendered output matches the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
