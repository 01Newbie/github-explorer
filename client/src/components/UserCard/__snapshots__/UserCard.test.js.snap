// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard UserCard snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      John Doe
    </h2>
    <p>
      john.doe@example.com
    </p>
    <div>
      <h3>
        Repositories:
      </h3>
      <div>
        <h4>
          Repo1
        </h4>
        <p>
          First repo
        </p>
        <p>
          Created on: 2024-01-01
        </p>
        <p>
          Last commit: 2024-01-10
        </p>
        <div>
          <h5>
            Last Commits:
          </h5>
          <ul>
            <li>
              Initial commit
            </li>
            <li>
              Updated README
            </li>
            <li>
              Fixed bugs
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
